actors/builtin/market/market_state.go

```diff
@@ -383,10 +380,10 @@ func dealProposalIsInternallyValid(rt Runtime, proposal ClientDealProposal) bool
 	// authenticity of the on-chain message publishing the deal.
 	sigVerified := rt.Syscalls().VerifySignature(proposal.ClientSignature, proposal.Proposal.Client, m)
 	if !sigVerified {
-		return false
+		return xerrors.Errorf("signature proposal invalid")
```

Comment:
FYI see https://github.com/filecoin-project/specs-actors/issues/126. I concur syscalls should return errors.